packages/x-date-pickers-pro/src/DateRangeCalendar/useDragRange.ts

```diff
@@ -173,14 +167,29 @@ const useDragRangeEvents = <TDate>({
 
   const handleTouchMove = useEventCallback((event: React.TouchEvent<HTMLButtonElement>) => {
     const target = resolveElementFromTouch(event);
-    if (!isDragging || !target) {
+    if (!target) {
       return;
     }
 
     const newDate = resolveDateFromTarget(target, utils, timezone);
     if (newDate) {
       setRangeDragDay(newDate);
     }
+
+    // this prevents initiating drag when user starts touchmove outside and then moves over a draggable element
+    const targetsAreIdentical = target === event.changedTouches[0].target;
+    if (!(targetsAreIdentical && isElementDraggable(newDate))) {
```

Comment:
I personally find the following easier to read:

```suggestion
    if (!targetsAreIdentical || !isElementDraggable(newDate)) {
```

Maybe it's just because I never write `!(A & B)` so my brain is not as fast as reading it

Reply:
its ok either way .. i do struggle with your approach as i am more used to `!(A && B)` (mentally). 🙇🏼

Reply:
:laughing: Yeah it's quite subjective